src/org/ssgwt/client/ui/form/spinner/DurationSpinner.java

```diff
@@ -523,6 +543,73 @@ public DurationSpinner(
+        
+        // Add blur handlers
+        hourValueBox.addBlurHandler(new BlurHandler() {
+            
+            /**
+             * This method will fire if focus is lost on the hour input field
+             * 
+             * @author Michael Barnard <michael.barnard@a24group.com>
+             * @since  3 August 2015
+             * 
+             * @param event - The blur event that made this code execute
+             */
+            @Override
+            public void onBlur(BlurEvent event) {
+                if (!readOnly) {
+                    try {
+                        int newHourValue = Integer.parseInt(hourValueBox.getValue());
+                        if (oldHourValue != newHourValue) {
+                            if (newHourValue > getHourSpinner().getMax()) {
+                                newHourValue = (int)getHourSpinner().getMax();
+                            }
+                            if (newHourValue < getHourSpinner().getMin()) {
+                                newHourValue = (int)getHourSpinner().getMin();
+                            }
+                            getHourSpinner().setValue(newHourValue, true);
+                            oldHourValue = newHourValue;
+                        }
+                    } catch (Exception e) {
+                        // This means the number entered was invalid... reverting back to the original
+                        getHourSpinner().setValue(oldHourValue, true);
+                    }
+                }
+            }
+        });
+        
+        minuteValueBox.addBlurHandler(new BlurHandler() {
+            
+            /**
+             * This method will fire if focus is lost on the minute input field
+             * 
+             * @author Michael Barnard <michael.barnard@a24group.com>
+             * @since  3 August 2015
+             * 
+             * @param event - The blur event that made this code execute
+             */
+            @Override
+            public void onBlur(BlurEvent event) {
+                if (!readOnly) {
+                    try {
+                        int newMinuteValue = Integer.parseInt(minuteValueBox.getValue());
+                        if (oldMinuteValue != newMinuteValue) {
+                            if (newMinuteValue > getMinuteSpinner().getMax()) {
+                                newMinuteValue = (int)getMinuteSpinner().getMax();
+                            }
+                            if (newMinuteValue < getMinuteSpinner().getMin()) {
+                                newMinuteValue = (int)getMinuteSpinner().getMin();
+                            }
+                            getMinuteSpinner().setValue(newMinuteValue, true);
+                            oldMinuteValue = newMinuteValue;
+                        }
+                    } catch (Exception e) {
+                        // This means the number entered was invalid... reverting back to the original
```

Comment:
Could you please make it clear that it is referring to the actual value being altered back to the original value.